### PR TITLE
Fix #655: Landscape Topic Overview

### LINKED ISSUE #655: Lowfi Landscape Topic Overview
Mock Link: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/4e4d2aec-fd30-4c20-91ed-daf434f47a89/TP-Info-

## Checklist
- [ ] Landscape lowfi mock
- [ ] Test cases

### PR DESCRIPTION
**Pull request by SayantanBanerjee16**
## Explanation

Fixes #655

Landscape mode created for the Topic Info fragment and necessary tests implemented

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt [hunk 1]
```diff
@@ -67,17 +69,23 @@ class TopicInfoFragmentTest {
   }
 
   @Test
-  @Ignore("Landscape not properly supported") // TODO(#56): Reenable once landscape is supported.
   fun testTopicInfoFragment_loadFragment_configurationChange_checkTopicName_isCorrect() {
     launchTopicActivityIntent(TEST_TOPIC_ID).use { scenario ->
       scenario.onActivity {  activity ->
         activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
       }
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt]: Update this orientation change step with : `onView(ViewMatchers.isRoot()).perform(orientationLandscape())`

File: app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt [hunk 2]
```diff
@@ -67,17 +69,23 @@ class TopicInfoFragmentTest {
...
-
       onView(withId(R.id.topic_name_text_view)).check(matches(withText(containsString(TOPIC_NAME))))
     }
   }
 
+  @Test
+  fun testTopicInfoFragment_loadFragment_configurationLandscape_isCorrect() {
+    launchTopicActivityIntent(TEST_TOPIC_ID).use {
+      onView(ViewMatchers.isRoot()).perform(OrientationChangeAction.orientationLandscape())
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt]: You can directly import it: 
``onView(ViewMatchers.isRoot()).perform(orientationLandscape())``

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt]: Once this is done, check there are no unused imports.

**Review by rt4914 — CHANGES_REQUESTED** [review]
There are few more suggestions:

1. Update your PR title to -> `Fix #655: Landscape Topic Overview`
2. Add one more test case which check that the imageview is not there in landscape mode. (NOTE: this would be tricky it do, so you might need to take help from other resources.)

**Comment by SayantanBanerjee16** [comment]
> There are few more suggestions:
> 
> 1. Update your PR title to -> `Fix #655: Landscape Topic Overview`
> 2. Add one more test case which check that the imageview is not there in landscape mode. (NOTE: this would be tricky it do, so you might need to take help from other resources.)

@rt4914 Done with all the changes. PTAL :)

File: app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt [hunk 3]
```diff
@@ -67,17 +69,29 @@ class TopicInfoFragmentTest {
...
+    launchTopicActivityIntent(TEST_TOPIC_ID).use {
+      onView(ViewMatchers.isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.topic_tabs_viewpager_container)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testTopicInfoFragment_loadFragment_configurationLandscape_imageViewNotDisplayed() {
+    launchTopicActivityIntent(TEST_TOPIC_ID).use {
+      onView(ViewMatchers.isRoot()).perform(orientationLandscape())
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt]: For better code clarity write it like this:
`onView(isRoot()).perform(orientationLandscape())`

Do this in other test cases too and accordingly remove unused imports.

**Review by rt4914 — CHANGES_REQUESTED** [review]
This is a nice implementation. Thanks. Just on small change.

File: app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt (hunk 3 above)
Comment (SayantanBanerjee16) [reply app/src/sharedTest/java/org/oppia/app/topic/info/TopicInfoFragmentTest.kt]: > For better code clarity write it like this:
> `onView(isRoot()).perform(orientationLandscape())`
> 
> Do this in other test cases too and accordingly remove unused imports.

@rt4914 Changes done. Do merge it and assign me a new issue. Thank you :)

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**